src/components/layouts/Layout.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from "react";
+import { Outlet } from "react-router";
+import { Toaster } from "sonner";
+import Searchbar from "../ui/Searchbar";
+import Sidebar from "../ui/Sidebar";
+import Topbar from "../ui/Topbar";
+import { useGetProfileInformationQuery } from "../../gql/graphql";
+import { useAppStore } from "../../store/store";
+
+const Layout = () => {
+  const isSidebarOpen = useAppStore((state) => state.isSidebarOpen);
+  const setIsSidebarOpen = useAppStore((state) => state.setIsSidebarOpen);  
+  const setUser = useAppStore((state) => state.setUser);
+  const { data, loading } = useGetProfileInformationQuery();
+
+  const handleSidebarToggle = () => {
+    setIsSidebarOpen(!isSidebarOpen);
+  };
+
+  useEffect(() => {
+    if (data?.profile) {
+      setUser(data?.profile);
+    } 
+  }, [data]);
```

Comment:
setUser is needed in the dependency array , please review if that affects your logic, in case yes try to fix it, is a good practice not ignore the dependencies here